src/main/java/com/google/sps/servlets/GroupsServlet.java

```diff
@@ -21,21 +26,45 @@
      */
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
-        // Hard coded data for now but will use data from datastore
-        // as groups are added to datastore
+        // Hard coded groups for now, but hopefully will use user-created groups from datastore
         // Photo by Hybrid on Unsplash 
         // (https://unsplash.com/@artbyhybrid?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText)
-        ImmutableMap<String, String> groupMap = ImmutableMap.of("groupName", "Black Lives Matter",
-            "groupImage", "https://images.unsplash.com/photo-1591622414912-34f2a8f8172e?ixlib=" + 
-            "rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80",
-            "groupDescription", "Group description would go here...");
-        ImmutableList<ImmutableMap<String, String>> groups = ImmutableList.of(groupMap);
+        Entity blmEntity = new Entity(GROUP_KIND, "123");
```

Comment:
I think creation of hardcoded entities should be somewhere else and not here, or it should check whether the groups were already created. Won't this try to create two groups every time you try to look at the groups there are?

Reply:
do you have any ideas about where I should create the sample groups? to my understanding, if I try to create a new entity with the same key, it just updates the already existing entity, but it would be better if it only had to create the entities once

Reply:
i think just call the groupsservlet, since i think you already have this code there. as for testing, just create the groups in junit with datastore and all that

Reply:
Since this is a big block of code to make the hard-coded groups, put it in a private helper function in this class. That way the reader doesn't have to read through and try to understand this chunk when they probably care more about what comes after.